lib/Form/Type/FieldType/BinaryBaseFieldType.php

```diff
@@ -32,12 +35,50 @@ public function buildForm(FormBuilderInterface $builder, array $options)
                 [
                     'label' => /** @Desc("File") */ 'content.field_type.binary_base.file',
                     'required' => $options['required'],
+                    'constraints' => [
+                        new Assert\File([
+                            'maxSize' => $this->getMaxUploadSize(),
+                        ]),
+                    ],
                 ]
             );
     }
 
+    public function buildView(FormView $view, FormInterface $form, array $options)
+    {
+        $view->vars['max_upload_size'] = $this->getMaxUploadSize();
+    }
+
     public function configureOptions(OptionsResolver $resolver)
     {
         $resolver->setDefaults(['translation_domain' => 'ezrepoforms_fieldtype']);
     }
+
+    private function getMaxUploadSize()
+    {
+        static $value = null;
+        if ($value === null) {
+            return $this->str2bytes(ini_get('upload_max_filesize'));
+        }
+
+        return $value;
+    }
+
+    private function str2bytes($str)
+    {
+        $str = strtoupper(trim($str));
+
+        $value = substr($str, 0, -1);
+        $unit = substr($str, -1);
+        switch ($unit) {
+            case 'G':
+                $value *= 1024;
```

Comment:
no break here?

Reply:
Yes, it's intentionally. To convert gigabytes to bytes we want to multiply a value by 1024^3, for M by 1024^2 and so on.  Ref: http://php.net/manual/en/faq.using.php#faq.using.shorthandbytes

Reply:
sorry, then i don't understand. if i put `1G` here will value be 1 * 1024 * 1024 * 1024?

wouldn't it be only `1 * 1024` as `case M` and `case K` will be false?

Reply:
> if i put 1G here will value be 1 * 1024 * 1024 * 1024?

yes

> wouldn't it be only 1 * 1024 as case M and case K will be false?

as you said above, there is no break, so it does not matter if case is false.

So this is why units are sorted by size here, so falltrue is taken advantage of.

Reply:
ok, understood now.